homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
+FROM jobs
+    INNER JOIN employees USING (job_id)
+GROUP BY job_title;
+
 --13. the country name, city, and number of those departments where at least 2 employees are working.
+SELECT country_name, city, COUNT(employee_id) AS employees_count
+FROM locations
+    INNER JOIN countries USING (country_id)
+    INNER JOIN departments USING (location_id)
+    INNER JOIN employees USING (department_id)
+GROUP BY country_name, city
+HAVING COUNT(employee_id) >= 2;
+
 --14. the employee ID, job name, number of days worked in for all those jobs in department 80.
+SELECT employee_id, job_title, end_date - start_date AS days_worked
+FROM job_history
+    INNER JOIN jobs USING (job_id)
+WHERE department_id = 80;
+
 --15. the name ( first name and last name ) for those employees who gets more salary than the employee whose ID is 163.
+SELECT (first_name || ' ' || last_name) AS employee_name
+FROM employees
+WHERE salary > (
+    SELECT salary
+    FROM employees
+    WHERE employee_id = 163
+    );
+
 --16. the employee id, employee name (first name and last name ) for all employees who earn more than the average salary.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name
+FROM employees
+WHERE salary > (
+    SELECT AVG(salary)
+    FROM employees
+);
+
 --17. the employee name ( first name and last name ), employee id and salary of all employees who report to Payam.
+SELECT (first_name || ' ' || last_name) AS emploee_name, employee_id, salary
+FROM employees
+WHERE manager_id = (
+    SELECT employee_id
+    FROM employees
+    WHERE first_name = 'Payam'
+);
+
 --18. the department number, name ( first name and last name ), job and department name for all employees in the Finance department.
+SELECT department_id, (first_name || ' ' || last_name) AS emploee_name, job_title, department_name
+FROM employees
+    INNER JOIN departments USING (department_id)
+    INNER JOIN jobs USING (job_id)
+WHERE department_name = 'Finance';
+
 --19. all the information of an employee whose id is any of the number 134, 159 and 183.
+SELECT *
+FROM employees
+WHERE employee_id IN (134, 159, 183);
+
 --20. all the information of the employees whose salary is within the range of smallest salary and 2500.
+SELECT *
+FROM employees
+WHERE salary BETWEEN (
+    SELECT MIN(salary)
+    FROM employees
+    ) AND 2500;
+
 --21. all the information of the employees who does not work in those departments where some employees works whose id within the range 100 and 200.
+SELECT *
+      FROM employees
+      WHERE department_id NOT IN (
+          SELECT department_id
+          FROM employees
+                   INNER JOIN departments USING (department_id)
+          WHERE employee_id BETWEEN 100 AND 200
+      );
+
 --22. all the information for those employees whose id is any id who earn the second highest salary.
+SELECT *
+FROM employees
+WHERE salary IN (SELECT MAX(salary)
+                 FROM employees
+                 WHERE salary != (SELECT MAX(salary)
```

Comment:
That will work too, but I guess the better choice here would be "<" instead of "!=".

Reply:
Done